src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAssociationManager.java

```diff
@@ -238,25 +238,24 @@ public AssociationToken createAssociationWithToken(final String token, final Reg
             throw new InvalidUserAssociationTokenException("The group token provided does not exist or is invalid.");
         }
 
-        // add owner association
-        if (!associationDatabase
-                .hasValidAssociation(lookedupToken.getOwnerUserId(), userGrantingPermission.getId())) {
-            associationDatabase.createAssociation(lookedupToken, userGrantingPermission.getId());
+        UserGroupDTO group = userGroupManager.getGroupById(lookedupToken.getGroupId());
+
+        if (lookedupToken.getGroupId() == null) {
+            throw new InvalidUserAssociationTokenException("The group token provided does not exist or is invalid.");
```

Comment:
Why is this check here rather than before the previous line? We could possibly skip a database access in the error case

Reply:
Actually, I think this should 100% be before the previous line given that you'll run into a null pointer exception otherwise :/

Reply:
On the earlier `null == lookedupToken` check you could change it to `null == lookedupToken || null == lookedupToken.getGroupId()`

Reply:
The (reverse of the) check was after the group lookup before, if you look at the old code. You're right, though, that it's not in the right place. I also think it's checking the wrong thing anyhow. Will go and take a detailed look! 🙌